packages/better-auth/src/plugins/jwt/sign.ts

```diff
@@ -55,23 +116,123 @@ export async function getJwtToken(
 		options?.jwks?.keyPairConfig?.alg ?? "EdDSA",
 	);
 
+	const jwt = new SignJWT(payload)
+		.setProtectedHeader({
+			alg: options?.jwks?.keyPairConfig?.alg ?? "EdDSA",
+			kid: key.id,
+			typ: "JWT",
+		})
+		.setExpirationTime(exp)
+		.setIssuer(iss ?? defaultIss)
+		.setAudience(aud ?? defaultAud);
+	const sub =
+		payload.sub ??
+		(await options?.jwt?.getSubject?.(ctx.context.session!)) ??
+		ctx.context.session?.user.id;
+	if (sub) jwt.setSubject(sub);
+	if (payload.iat) jwt.setIssuedAt(iat);
+	if (payload.nbf) jwt.setNotBefore(payload.nbf);
+	if (payload.jti) jwt.setJti(payload.jti);
+	return await jwt.sign(privateKey);
+}
+
+/**
+ * Signs a payload in jwt format.
+ *
+ * @param ctx - endpoint context
+ * @param payload - payload to sign
+ */
+// Plugin exportable
+export async function signJwt(
+	ctx: GenericEndpointContext,
+	payload: JWTPayload,
+	safeguards?: JwtSignSafeguards,
+) {
+	const options = getJwtPlugin(ctx.context).options;
+	return signJwtPayload(ctx, payload, options, safeguards);
+}
+
+/**
+ * Backwards compatable version of signJwt
+ *
+ * @deprecated - prefer signJwt to prevent option injection
```

Comment:
Actually does not prevent option injection because updating the `ctx.context` directly would do the same.